lib/minify.js

```diff
@@ -291,11 +288,21 @@ async function minify(files, options, _fs_module) {
         toplevel.print(stream);
         result.code = stream.get();
         if (options.sourceMap) {
-            if(options.sourceMap.asObject) {
-                result.map = options.format.source_map.get().toJSON();
-            } else {
-                result.map = options.format.source_map.toString();
-            }
+            Object.defineProperty(result, "map", {
+                configurable: true,
+                enumerable: true,
+                get() {
+                    const map = options.format.source_map.getEncoded();
+                    return (result.map = options.sourceMap.asObject ? map : JSON.stringify(map));
```

Comment:
Encoded maps require serialization from decoded form into an encoded VLQ string. I've placed this behind a getter so that it's not performed unless required by the dev.

Reply:
Unrelated suggestion:

I looked at the implementation underneath getEncoded [here](https://github.com/jridgewell/gen-mapping/blob/main/src/gen-mapping.ts#L244-L250) and I wonder if a JSON string rendition of it could be optimized.

VLQ strings are guaranteed to not contain `\` or `"` so for large maps it may be more memory efficient to generate the JSON manually.

`'{"files": (...), "mappings": "' + encode(decoded.mappings) + '"}'`

I'm making the assumption that V8 would make this more memory efficient by not copying the encoded string into the JSON, but instead making the output JSON into a rope containing 3 strings.